app/src/test/kotlin/com/k0d4black/theforce/activities/SearchActivityUnitTest.kt

```diff
@@ -40,11 +40,6 @@ class SearchActivityUnitTest {
 
     @Test
     fun `validate search menu is displayed`() {
-        searchActivity =
-            Robolectric.buildActivity(SearchActivity::class.java)
-                .create()
-                .visible()
-                .get()
         val menu = Shadows.shadowOf(searchActivity).optionsMenu
         val menuItem = menu.findItem(R.id.action_search)
         assertThat(menuItem.isVisible)
```

Comment:
As for this test, for it to pass it needs the activity to be visible so as to access the menu.

Changing the ```.resume()``` to ```.visible()``` while building the activity in the ```@Before``` method should solve it.

Reply:
after I changed the parent class of `TheForceTestApplication `solves this but it only passes while runs independently from previous test case, it still fails while running `SearchActivityUnitTest `class

Reply:
AFAIS from logcat it tries to start dependencies from project`s application class, this seems weird to me 😕 

`java.lang.IllegalStateException: A KoinContext is already started
``at com.k0d4black.theforce.TheForceApplication.onCreate(TheForceApplication.kt:18)`

Reply:
It seems awkward , for the ```SearchActivityUnitTest``` you had solved the redundant reassigning of search activity. The only change it would have required was on the ```@Before```
```kotlin
    @Before
    fun setup() {
        searchActivity =
            Robolectric.buildActivity(SearchActivity::class.java)
                .create()
                .visible()
                .get()
    }
```

Reply:
I just kept this initialization only in setup but it sill gives same error 😞

Reply:
At least we have narrowed down the problem to just this test.Everything else seems fine on CI

Reply:
exactly

Reply:
To Resolve this have the test extent the ```AutoCloseKoinTest``` that way Koin will handle closing of Koin after every test and there will be a fresh context for each test.

The end result should be something of the sort

```kotlin
@RunWith(AndroidJUnit4::class)
@Config(sdk = [Build.VERSION_CODES.O_MR1])
class SearchActivityUnitTest:AutoCloseKoinTest(){

private lateinit var searchActivity: SearchActivity

    @Before
    fun setup() {
        searchActivity =
            Robolectric.buildActivity(SearchActivity::class.java)
                .create()
                .visible()
                .get()
    }

//The Test methods
}
```

Reply:
thanks this is just another TIL 😄